src/LDNS.xs

```diff
@@ -2543,6 +2542,34 @@ rr_nsec3_covers(obj,name)
     OUTPUT:
         RETVAL
 
+SV *
+rr_nsec3_hash_name(obj,name)
+    Zonemaster::LDNS::RR::NSEC3 obj;
+    const char *name;
+    INIT:
+        /* Sanity test on owner name */
+        if (ldns_dname_label_count(ldns_rr_owner(obj)) == 0)
+            XSRETURN_UNDEF;
+    CODE:
+    {
+        ldns_rdf *hashed;
+        ldns_rdf *dname;
+
+        dname = ldns_rdf_new_frm_str(LDNS_RDF_TYPE_DNAME, name);
+        if (!dname)
+            XSRETURN_UNDEF;
+
+        hashed = ldns_nsec3_hash_name_frm_nsec3(obj, dname);
+
+        if (!hashed || ldns_rdf_size(hashed) < 1 ) {
+            XSRETURN_UNDEF;
+        }
+
+        RETVAL = newSVpvn((char*)(ldns_rdf2str(hashed)), ldns_rdf_size(hashed) - 2 );
```

Comment:
Not quite: `ldns_rdf2str()` returns a NUL-terminated string, but we need to free the memory ourselves when done.

```suggestion
        char *hash_str = ldns_rdf2str(hashed);
        RETVAL = newSVpv(ldns_rdf2str(hashed), 0);
        free(hash_str);
```